packages/next/client/head-manager.ts

```diff
@@ -89,7 +89,9 @@ function updateElements(
 
 export default function initHeadManager(initialHeadEntries: HeadEntry[]) {
   const headEl = document.getElementsByTagName('head')[0]
-  const elements = new Set<Element>(headEl.children)
+  // Some browsers like Edge 16 don't support constructing `Set` with an `HTMLCollection`,
+  // so we need to first convert it using `Array.from`.
+  const elements = new Set<Element>(Array.from(headEl.children))
```

Comment:
For posterity, this won't fix: #17721 per @timneutkens comments on it not working for client-side navigation.

Reply:
Could you elaborate on your comment, please? Are you just referring to custom components not being client-side rendered?

Reply:
Referring to https://github.com/vercel/next.js/issues/17721#issuecomment-709487069 which is the "previous" behavior in 9.5.3, which I think this PR will get us back to that, but not solve the issue. But you've explained further concerns about this in #17721.